google/cloud/bigquery/schema.py

```diff
@@ -206,6 +240,22 @@ def __init__(
             self._properties["rangeElementType"] = {"type": range_element_type}
         if isinstance(range_element_type, FieldElementType):
             self._properties["rangeElementType"] = range_element_type.to_api_repr()
+        if rounding_mode is not None:
+            if isinstance(rounding_mode, enums.RoundingMode):
+                rounding_mode = rounding_mode.name
```

Comment:
I think if we do the `RoundingMode(str, enum.Enum)` trick, this logic should be unnecessary.

Would be good to double check in some unit tests that `json.dumps(field.to_api_repr())` works when passing in one of these enums if we do remove this if statement.

Reply:
With the changes to the `Rounding_Mode` enum and our existing test suite, we do have a test that ensures given an enum representation, we get the correct JSON output when calling `to_api_repr()`. Specifically, see `test_to_api_repr()`.